server/auth/auth.go

```diff
@@ -1131,6 +1190,13 @@ func (auth *AuthManager) handleConnect(conn comms.Link, msg *msgjson.Message) *m
 		client.suspended = true
 		auth.storage.CloseAccount(user, account.FailureToAct)
 		log.Debugf("Suspended account %v (score = %d) connected.", acctInfo.ID, score)
+	} else if score < int32(auth.banScore) && !open {
+		if err = auth.Unban(user); err == nil {
+			log.Warnf("Restoring suspended account %v (score = %d).", acctInfo.ID, score)
+		} else {
+			log.Errorf("Failed to restore suspended account %v (score = %d): %v.",
+				acctInfo.ID, score, err)
+		}
```

Comment:
`banScore` is an operator setting, but even so this branch would have automated account restoration on reconnect in the event that this bug manifested.